docs/projects/version-control/creating-and-deploying-releases-version-controlled-project.md

```diff
@@ -18,12 +18,17 @@ When you create a release via the UI, you must specify a branch name. Octopus wi
 
 ### Creating a release from a build server plug-in
 
-We have added two new fields to our standard integrations - TeamCity, Azure DevOps, Jenkins, GitHub Actions, and Bamboo.
+Octopus does not guess or auto-populate the commit or branch when creating a release from a build-server plug-in. Instead, to provide this information, we have added two new fields to our standard integrations - TeamCity, Azure DevOps, Jenkins, GitHub Actions, and Bamboo.
 
-* Git Reference - the user-friendly alias for a commit hash.
+* Git Reference - a user-friendly alias for a commit hash. This is typically a branch name or tag.
 * Git Commit - the commit SHA-1 hash.
 
-When the app is built in a different repository to the Octopus project, Octopus does not guess or auto-populate the commit or branch from which you want to create the release. Also, if the app and the Octopus project are in the same repository, the head of that branch could have moved forward from what is expected. It is highly recommended that you provide the commit and not just the branch in both cases.
+The use of these fields can change depending on where the version-controlled project's OCL files are stored:
+
+1. If the OCL files are stored in the **same repository** as the application(s) being built, it's likely that a specific commit that relates to any artifacts created as part of the build itself should be used when creating the release. In this scenario, you should provide both the Git Reference and Git Commit hash of the executing build. This ensures that the release will use the correct version of the project, and won't include any potential changes made to the `HEAD` of the branch *before* the build has completed.
+2. If the OCL files are stored in a **different repository** than the application(s) being built, a specific branch or tag can identify which version of the project to use when creating the release. In this case, you would provide the Git Reference where the OCL files are stored, and not the Git Commit hash. e.g. Use the `main` branch, regardless of the location of the repository where the application(s) are being built as they are different.
+
+It is highly recommended that you provide the commit and not just the branch in both cases.
```

Comment:
Point 2 nicely covers how and why you'd point to just a branch, and then we are saying users still probably shouldn't. Perhaps we can just drop this line all together?

Reply:
I agree, change made if @harrisonmeister also agrees!

Reply:
Works for me